lib/reporter.js

```diff
@@ -0,0 +1,45 @@
+var chalk = require('chalk');
+var _ = require('lodash');
+var defaultFormatter = require('./defaultFormatter');
+
+module.exports = function(opts) {
+  var options = opts || {};
+
+  var formatter = options.formatter || defaultFormatter();
+
+  return function(css, result) {
+    var messagesToLog = (!options.plugins || !options.plugins.length)
+      ? result.messages
+      : result.messages.filter(function(message) {
+          if (options.plugins.indexOf(message.plugin) !== -1) {
```

Comment:
It's the bitwise NOT operator. It basically will shift `-1` to `0` and `0` to `-1`; if the value is found, it will evaluate to true:

```js
var str = 'ab';
console.log(~str.indexOf('ab')); // -1 (true)
```

If the value is not found:

```js
var str = 'cd';
console.log(~str.indexOf('ab')) // 0 (false)
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators#Bitwise_NOT